### PR TITLE
Segregate Even and Odd numbers in a Linked List

### PR DESCRIPTION
**Pull request by radhikaaojha**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

Segregate Even and Odd nodes in a Linked List using C.

### Checklist

- [x] I've read the contribution guidelines.

- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#6816

### CODE REVIEW

**Comment by radhikaaojha** [comment]
@atarax665 Please Review my PR

**Comment by atarax665** [comment]
Please fill the PR template and mention the details of your contribution, also mention the linked issue as here #6816.

File: C/ds/SegregateOddEven.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	{
+		/* If the data of node is ODD */
+    	if (temp->data % 2 != 0)	
+    	{
+      		if (temp == header)		//If the first node is odd,set prev as header and then update header to the next node 
+      		{
+        		prev = header;
+        		header = header->next;
+      		}
+      		next_node = temp->next;		//Store value of current odd node's next in next_node
```
Comment (atarax665) [inline C/ds/SegregateOddEven.c]: Move comment above the line of code.

File: C/ds/SegregateOddEven.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	/* If the last node is ODD */
+    if (p1->data % 2 != 0)		//p1 points to the last node here
+  	{
+    	next_node = temp->next;
+    	prev->next = next_node;
+    	last->next = temp;
+    	temp->next = NULL;
+    	last = temp;
+  	}
+}
```
Comment (atarax665) [inline C/ds/SegregateOddEven.c:142]: Add space/time complexity and sample input/output. Please refer to any recently merged code to get a better insight.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by anushkrishnav** [comment]
Please solve the conflict issue

**Comment by radhikaaojha** [comment]
I've done the requested changes.Please review my PR @atarax665

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by ankitaggarwal23** [comment]
@all-contributors bot please add @radhikaaojha for code

**Pull request merged**